tools/isobuild/compiler-plugin.js

```diff
@@ -1295,20 +1306,25 @@ export class PackageSourceBatch {
         watchSet: batch.unibuild.watchSet,
       });
 
-      scanner.addInputFiles(map.get(name).files);
+      await scanner.addInputFiles(map.get(name).files);
+
+      cancelToken.throwIfRequested();
 
       if (batch.useMeteorInstall) {
-        scanner.scanImports();
+        await scanner.scanImports();
```

Comment:
The `ImportScanner#scanImports` method is an expensive chunk of the build process that doesn't currently have any `cancelToken.throwIfInterrupted()` calls within it.